clients/src/main/java/org/apache/kafka/clients/ApiVersions.java

```diff
@@ -64,4 +64,8 @@ public synchronized byte maxUsableProduceMagic() {
         return maxUsableProduceMagic;
     }
 
+    // check if all nodes are ZK Migration ready
+    public boolean isAllNodeZkMigrationReady() {
```

Comment:
Can we move this logic into QuorumFeatures? That would allow us to verify that all the expected controllers are present in the ApiVersions (in addition to checking the `zkMigrationEnabled` flag)

Reply:
Agree! Moving to `QuorumFeatures` now, and add tests to test the non-controller node case. Thanks.